hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/NewHoodieParquetFileFormat.scala

```diff
@@ -72,7 +72,7 @@ class NewHoodieParquetFileFormat(tableState: Broadcast[HoodieTableState],
   private var supportBatchCalled = false
   private var supportBatchResult = false
   override def supportBatch(sparkSession: SparkSession, schema: StructType): Boolean = {
-    if (!supportBatchCalled) {
+    if (!supportBatchCalled || supportBatchResult) {
```

Comment:
Please clarify, if it's MOR table `supportBatch` result will always be false. Then we can hit the same issue right? `supportBatchCalled` can be true on multiple invocations of file format.

Reply:
My understanding is that the issue only happens on COW tables.